source/grid/grid_generator.cc

```diff
@@ -3625,6 +3625,25 @@ namespace GridGenerator
     GridGenerator::merge_triangulations(
       tria_without_cylinder, cylinder_tria, tria, vertex_tolerance, true);
 
+    // Move the vertices in the middle of the faces of cylinder_tria slightly
+    // to give a better mesh quality. We choose to move them such that the
+    // distance between (0.05 + shell_region_width) and 0.1 is at least half
+    // as long as the remaining length to the boundary.
+    const double shifted_point =
```

Comment:
This is an awkward naming of the variable since it's not actually a point, and it's not the shifted location either but really just the "shift" if I see correctly how you're using it below.